crates/core/tedge_agent/src/state_repository/error.rs

```diff
@@ -10,6 +12,12 @@ pub enum StateError {
     #[error(transparent)]
     FromIo(#[from] std::io::Error),
 
+    #[error("Could not read tedge-agent state from file {repo_path}: {source}")]
```

Comment:
When starting a container with only `tedge` and `tedge-agent` installed, the following error message showed on startup:

`Fail to read tedge-agent state: Could not read tedge-agent state from file /etc/tedge/.agent/restart-current-operation: No such file or directory (os error 2)`

Reply:
Beware that a related change has been merged recently: https://github.com/thin-edge/thin-edge.io/pull/2434

Reply:
I think the problem with this message, before and after #2434, is that it doesn't say the path of the file to which it refers. In the normal case when the file doesn't exist maybe it doesn't matter, but for other cases, like for permissions denied, I think it should absolutely say what file could not be opened.

I changed the offending commit and these are the examples new error messages:

- when file doesn't exist:
  ```
  2023-11-09T19:10:10.934643478Z  INFO tedge_agent::restart_manager::actor: tedge-agent state file `/etc/tedge/.agent/restart-current-operation` is not present

  ```

- when file can't be read due to bad permissions
  ```
  2023-11-09T19:09:36.765052563Z ERROR tedge_agent::restart_manager::actor: Could not read state from file `/etc/tedge/.agent/restart-current-operation`: Permission denied (os error 13)
  ```

- when file can't be correctly parsed
  ```
  2023-11-09T19:07:49.7470124Z ERROR tedge_agent::restart_manager::actor: State file `/etc/tedge/.agent/restart-current-operation` contains invalid syntax: TOML parse error at line 1, column 6
    |
  1 | file content
    |      ^
  expected `.`, `=`
  ```

We'll need to make sure to attach enough context to errors in a more systematic way. When I'll have time after 1.0 I'll create a dedicated issue where I'll talk about sinks for our errors (MQTT `te/errors` topic, logs, smartrest messages), how to present them better with miette, and how to better organise them and attach necessary context.

Reply:
> We'll need to make sure to attach enough context to errors in a more systematic way. When I'll have time after 1.0 I'll create a dedicated issue where I'll talk about sinks for our errors (MQTT te/errors topic, logs, smartrest messages), how to present them better with miette, and how to better organise them and attach necessary context.

Yes, please. Work in pair with @jarhodes314 who has also strong opinion and insight on that matter.
For 1.0, we can have local improvements like this one here and there. But as you said, we have to wait post 1.0 for broader improvements.